src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -498,7 +511,8 @@ async def get_repository_pex(
             ),
             interpreter_constraints=interpreter_constraints,
             platforms=request.platforms,
-            additional_args=request.additional_lockfile_args,
+            complete_platforms=request.complete_platforms,
+            additional_args=request.additional_args + request.additional_lockfile_args,
```

Comment:
Here and below on line 604 were bugs I fixed along the way.

The `request.additional_args` being dropped on the floor meant, for one, `resolve_local_platforms=True` in a `pex_binary` target was being dropped on the floor. I debugged this when testing the feature out on `src/python/pants/bin:pants` using a few `complete_platforms` which necessitated `resolve_local_platforms` since we have a dev dep on the pycharm debugger which is sdist only.

Reply:
This turns out to have been untrue. My state-tracking was broken. Test added that resolve_local_platforms works and these changes reverted.